src/arrayMethodJoin.js

```diff
@@ -4,8 +4,10 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    return String(this.length === 0 ? ''
+      : this.map(el => el === undefined || el === null ? '' : el)
+        .reduce((sum, curent) => sum + String(separator) + curent));
```

Comment:
I'd recommend using the next pattern for splitting ternary to separate lines:
```
condition
  ? ... code if true
  : ... code if false
```

Reply:
It is not clear enough why do you initialize a string with `String` constructor. `separator` is already `string`. I think there is no need to use `String` object

Reply:
The variable `sum` in `reduce` is not related to the current case. It's better to use a general definition like `accumulator` or, for the local case, `result`, `string`, `resultString` or smth else. `sum` is more for numbers